mono/metadata/sgen-mono.c

```diff
@@ -1350,14 +1350,6 @@ create_allocator (int atype, gboolean slowpath)
 		mono_mb_emit_byte (mb, MONO_CEE_ADD);
 		mono_mb_emit_ldarg (mb, 1);
 		mono_mb_emit_byte (mb, MONO_CEE_STIND_I4);
-		/* s->chars [len] = 0; */
-		mono_mb_emit_ldloc (mb, p_var);
-		mono_mb_emit_ldloc (mb, size_var);
-		mono_mb_emit_icon (mb, 2);
-		mono_mb_emit_byte (mb, MONO_CEE_SUB);
-		mono_mb_emit_byte (mb, MONO_CEE_ADD);
-		mono_mb_emit_icon (mb, 0);
-		mono_mb_emit_byte (mb, MONO_CEE_STIND_I2);
```

Comment:
Say we allocate a 20-byte string object (8-byte vtable pointer + 6 double-byte chars). At https://github.com/mono/mono/blob/alexischr/remove-managed-alloc-string-zeroing/mono/metadata/sgen-mono.c#L1245 size_var is aligned-up to 24, so the above would zero out offsets 22 & 23, not 20 & 21.

Also it's my understanding that we don't have to worry about null termination with SGen as allocated memory is always zeroed. So I removed instead of fixing.